python/az/aro/azext_aro/_validators.py

```diff
@@ -118,6 +118,26 @@ def validate_pull_secret(namespace):
         raise InvalidArgumentValueError("Invalid --pull-secret.") from e
 
 
+def validate_outbound_type(namespace):
+    outbound_type = getattr(namespace, 'outbound_type')
+    if outbound_type not in {'UserDefinedRouting', 'Loadbalancer', None}:
+        raise InvalidArgumentValueError('Invalid --outbound-type: must be "UserDefinedRouting" or "Loadbalancer"')
+
+    ingress_visibility = getattr(namespace, 'ingress_visibility')
+    apiserver_visibility = getattr(namespace, 'apiserver_visibility')
+
+    if (outbound_type == 'UserDefinedRouting' and
+            is_visibility_public(ingress_visibility) or
+            is_visibility_public(apiserver_visibility)):
```

Comment:
I think this is bugged, needs some `()` here to wrap the `or` condition (that is, I'm not sure why tests are passing? Maybe `getattr` is returning `""` instead of `None`? I'd recommend a debug session to confirm...). This is a result of Python order of operations on `and` vs `or`.

For example, this can result in a:

```
if False and True or True
```

which is equivalent to
```
if (False and True) or True
```

Basically, any time they set public API, they'll pass this `if`. So, the user doesn't set UserDefinedRouting, then we tell them, "you can't use that!"

Reply:
Python evaluates these expressions lazily, if the first expression is not UserDefinedRouting then the rest of the expression will not be evaluated.  If the first expression is True, then the next expression is evaluated. If that next expression is False then the next is evaulated.  So either way here we are going to encounter the exception if a user passes in UserDefinedRouting and if any visibility is public.